awswrangler/distributed/datasources/pandas_text_datasource.py

```diff
@@ -58,35 +77,128 @@ def _read_stream(  # type: ignore  # pylint: disable=arguments-differ
     def _read_file(self, f: pyarrow.NativeFile, path: str, **reader_args: Any) -> pd.DataFrame:
         raise NotImplementedError()
 
-    def _write_block(  # type: ignore  # pylint: disable=signature-differs
+    def do_write(  # type: ignore  # pylint: disable=arguments-differ
         self,
-        f: pyarrow.NativeFile,
+        blocks: List[ObjectRef[pd.DataFrame]],
+        metadata: List[BlockMetadata],
+        path: str,
+        dataset_uuid: str,
+        filesystem: Optional[pyarrow.fs.FileSystem] = None,
+        try_create_dir: bool = True,
+        open_stream_args: Optional[Dict[str, Any]] = None,
+        block_path_provider: BlockWritePathProvider = DefaultBlockWritePathProvider(),
+        write_args_fn: Callable[[], Dict[str, Any]] = lambda: {},
+        _block_udf: Optional[Callable[[pd.DataFrame], pd.DataFrame]] = None,
+        ray_remote_args: Optional[Dict[str, Any]] = None,
+        boto3_session: Optional[_utils.Boto3PrimitivesType] = None,
+        s3_additional_kwargs: Optional[Dict[str, str]] = None,
+        **write_args,
+    ) -> List[ObjectRef[WriteResult]]:
+        """Create and return write tasks for a file-based datasource."""
+        _write_block_to_file = self._write_block
+
+        if open_stream_args is None:
+            open_stream_args = {}
+
+        if ray_remote_args is None:
+            ray_remote_args = {}
+
+        def write_block(write_path: str, block: pd.DataFrame) -> str:
+            _logger.debug("Writing %s file.", write_path)
+            if _block_udf is not None:
+                block = _block_udf(block)
+
+            with open_s3_object(
+                path=write_path,
+                mode=write_args.get("mode"),
+                use_threads=False,
+                s3_additional_kwargs=s3_additional_kwargs,
+                boto3_session=boto3_session,
+                encoding=write_args.get("encoding"),
+                newline=write_args.get("newline"),
+            ) as f:
+                _write_block_to_file(
+                    f,
+                    PandasBlockAccessor(block),
+                    writer_args_fn=write_args_fn,
+                    **write_args,
+                )
+                return write_path
+
+        write_block_fn = cached_remote_fn(write_block).options(**ray_remote_args)
+
+        file_format = self._file_format()
+        write_tasks = []
+        if not block_path_provider:
```

Comment:
This check is already done in `_write_text` L148 so one of the two is redundant. If we go with this one we should have it in the ParquetDatasource class too

Reply:
You're right, good catch. I'll remove it from here for consistency's sake. I'd rather see this defined explicitely when we initialize a data source.